### PR TITLE
Fully test and annotate MAM preferences

### PR DESCRIPTION
**Pull request by JanuszJakubiec**
Testing and fixing any issues with muc_mam_prefs. The code was not tested before, so I've made some cosmetic changes. I've also changed the handle_set_prefs function, to prevent saving unusable entries to the database.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/215419) / elasticsearch_and_cassandra_mnesia / d811cda7e58fa56a2841c062ec808279c09173ae
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4265/215419/elasticsearch_and_cassandra_mnesia.26.1.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4265/215419/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%40ab0d3152735d.2024-04-15_07.22.17/index.html)
**OK: 439** / **Failed: 4** / **User-skipped: 41** / Auto-skipped: 0 
<details><summary>mam_SUITE:cassandra_muc_all:muc_prefs_cases:muc_prefs_set_cdata_request</summary>


```erlang
{error,
  {function_clause,
    [{mam_helper,parse_children_prefs_iq,
       [{xmlel,<<"error">>,
          [{<<"code">>,<<"501">>},{<<"type">>,<<"cancel">>}],
          [{xmlel,<<"feature-not-implemented">>,
             [{<<"xmlns">>,
             <<"urn:ietf:params:xml:ns:xmpp-stanzas">>}],
             []},
           {xmlel,<<"text">>,
             [{<<"xmlns">>,
             <<"urn:ietf:params:xml:ns:xmpp-stanzas">>}],
             [{xmlcdata,<<"From mod_mam_muc">>}]}]},
        {prefs_result_iq,undefined,[],[]}],
       []},
     {lists,foldl,3,[{file,"lists.erl"},{line,1594}]},
     {mam_SUITE,'-muc_prefs_set_cdata_request/1-fun-0-',2,
       [{file,"/home/circleci/project/big_tests/tests/mam_SUITE.erl"},
        {line,3383}]},
     {escalus_story,story,4,
       [{file,
          "/home/circleci/project/big_tests/_build/default/lib/escalus/src/escalus_story.erl"},
        {line,72}]},
     {muc_helper,story_with_room,4,
       [{file,"/home/circleci/project/big_tests/tests/muc_helper.erl"},
        {line,318}]},
     {test_server,ts_tc,3,[{file,"test_server.erl"},{line,1793}]},
     {test_server,run_test_case_eval1,6,
       [{file,"test_server.erl"},{line,1302}]},
     {test_server,run_test_case_eval,9,
       [{file,"test_server.erl"},{line,1234}]}]}}
```

[Report log](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4265/215419/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%40ab0d3152735d.2024-04-15_07.22.17/big_tests.tests.mam_SUITE.logs/run.2024-04-15_07.22.18/mam_suite.muc_prefs_set_cdata_request.8962.html)
</details>
<details><summary>mam_SUITE:cassandra_muc_all:muc_prefs_cases:muc_prefs_set_request</summary>


```erlang
{error,
  {function_clause,
    [{mam_helper,parse_children_prefs_iq,
       [{xmlel,<<"error">>,
          [{<<"code">>,<<"501">>},{<<"type">>,<<"cancel">>}],
          [{xmlel,<<"feature-not-implemented">>,
             [{<<"xmlns">>,
             <<"urn:ietf:params:xml:ns:xmpp-stanzas">>}],
             []},
           {xmlel,<<"text">>,
             [{<<"xmlns">>,
             <<"urn:ietf:params:xml:ns:xmpp-stanzas">>}],
             [{xmlcdata,<<"From mod_mam_muc">>}]}]},
        {prefs_result_iq,undefined,[],[]}],
       []},
     {lists,foldl,3,[{file,"lists.erl"},{line,1594}]},
     {mam_SUITE,'-muc_prefs_set_request/1-fun-0-',3,
       [{file,"/home/circleci/project/big_tests/tests/mam_SUITE.erl"},
        {line,3311}]},
     {escalus_story,story,4,
       [{file,
          "/home/circleci/project/big_tests/_build/default/lib/escalus/src/escalus_story.erl"},
        {line,72}]},
     {muc_helper,story_with_room,4,
       [{file,"/home/circleci/project/big_tests/tests/muc_helper.erl"},
        {line,318}]},
     {test_server,ts_tc,3,[{file,"test_server.erl"},{line,1793}]},
     {test_server,run_test_case_eval1,6,
       [{file,"test_server.erl"},{line,1302}]},
     {test_server,run_test_case_eval,9,
       [{file,"test_server.erl"},{line,1234}]}]}}
```

[Report log](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4265/215419/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%40ab0d3152735d.2024-04-15_07.22.17/big_tests.tests.mam_SUITE.logs/run.2024-04-15_07.22.18/mam_suite.muc_prefs_set_request.6915.html)
</details>
<details><summary>mam_SUITE:elasticsearch_muc_all:muc_prefs_cases:muc_prefs_set_cdata_request</summary>


```erlang
{error,
  {function_clause,
    [{mam_helper,parse_children_prefs_iq,
       [{xmlel,<<"error">>,
          [{<<"code">>,<<"501">>},{<<"type">>,<<"cancel">>}],
          [{xmlel,<<"feature-not-implemented">>,
             [{<<"xmlns">>,
             <<"urn:ietf:params:xml:ns:xmpp-stanzas">>}],
             []},
           {xmlel,<<"text">>,
             [{<<"xmlns">>,
             <<"urn:ietf:params:xml:ns:xmpp-stanzas">>}],
             [{xmlcdata,<<"From mod_mam_muc">>}]}]},
        {prefs_result_iq,undefined,[],[]}],
       []},
     {lists,foldl,3,[{file,"lists.erl"},{line,1594}]},
     {mam_SUITE,'-muc_prefs_set_cdata_request/1-fun-0-',2,
       [{file,"/home/circleci/project/big_tests/tests/mam_SUITE.erl"},
        {line,3383}]},
     {escalus_story,story,4,
       [{file,
          "/home/circleci/project/big_tests/_build/default/lib/escalus/src/escalus_story.erl"},
        {line,72}]},
     {muc_helper,story_with_room,4,
       [{file,"/home/circleci/project/big_tests/tests/muc_helper.erl"},
        {line,318}]},
     {test_server,ts_tc,3,[{file,"test_server.erl"},{line,1793}]},
     {test_server,run_test_case_eval1,6,
       [{file,"test_server.erl"},{line,1302}]},
     {test_server,run_test_case_eval,9,
       [{file,"test_server.erl"},{line,1234}]}]}}
```

[Report log](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4265/215419/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%40ab0d3152735d.2024-04-15_07.22.17/big_tests.tests.mam_SUITE.logs/run.2024-04-15_07.22.18/mam_suite.muc_prefs_set_cdata_request.21250.html)
</details>
<details><summary>mam_SUITE:elasticsearch_muc_all:muc_prefs_cases:muc_prefs_set_request</summary>


```erlang
{error,
  {function_clause,
    [{mam_helper,parse_children_prefs_iq,
       [{xmlel,<<"error">>,
          [{<<"code">>,<<"501">>},{<<"type">>,<<"cancel">>}],
          [{xmlel,<<"feature-not-implemented">>,
             [{<<"xmlns">>,
             <<"urn:ietf:params:xml:ns:xmpp-stanzas">>}],
             []},
           {xmlel,<<"text">>,
             [{<<"xmlns">>,
             <<"urn:ietf:params:xml:ns:xmpp-stanzas">>}],
             [{xmlcdata,<<"From mod_mam_muc">>}]}]},
        {prefs_result_iq,undefined,[],[]}],
       []},
     {lists,foldl,3,[{file,"lists.erl"},{line,1594}]},
     {mam_SUITE,'-muc_prefs_set_request/1-fun-0-',3,
       [{file,"/home/circleci/project/big_tests/tests/mam_SUITE.erl"},
        {line,3311}]},
     {escalus_story,story,4,
       [{file,
          "/home/circleci/project/big_tests/_build/default/lib/escalus/src/escalus_story.erl"},
        {line,72}]},
     {muc_helper,story_with_room,4,
       [{file,"/home/circleci/project/big_tests/tests/muc_helper.erl"},
        {line,318}]},
     {test_server,ts_tc,3,[{file,"test_server.erl"},{line,1793}]},
     {test_server,run_test_case_eval1,6,
       [{file,"test_server.erl"},{line,1302}]},
     {test_server,run_test_case_eval,9,
       [{file,"test_server.erl"},{line,1234}]}]}}
```

[Report log](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4265/215419/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%40ab0d3152735d.2024-04-15_07.22.17/big_tests.tests.mam_SUITE.logs/run.2024-04-15_07.22.18/mam_suite.muc_prefs_set_request.21122.html)
</details>

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/217970) / elasticsearch_and_cassandra_mnesia / c3b2e29bda843776e128ed51b5bb6c5e99fc8947
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4265/217970/elasticsearch_and_cassandra_mnesia.26.1.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4265/217970/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%40eb565858c58d.2024-04-30_06.48.16/index.html)
**OK: 439** / **Failed: 10** / **User-skipped: 41** / Auto-skipped: 0 
<details><summary>mam_SUITE:cassandra_muc_all:muc_prefs_cases:muc_prefs_set_cdata_request</summary>


```erlang
{error,
  {function_clause,
    [{mam_helper,parse_children_prefs_iq,
       [{xmlel,<<"error">>,
          [{<<"code">>,<<"501">>},{<<"type">>,<<"cancel">>}],
          [{xmlel,<<"feature-not-implemented">>,
             [{<<"xmlns">>,
             <<"urn:ietf:params:xml:ns:xmpp-stanzas">>}],
             []},
           {xmlel,<<"text">>,
             [{<<"xmlns">>,
             <<"urn:ietf:params:xml:ns:xmpp-stanzas">>}],
             [{xmlcdata,<<"From mod_mam_muc">>}]}]},
        {prefs_result_iq,undefined,[],[]}],
       []},
     {lists,foldl,3,[{file,"lists.erl"},{line,1594}]},
     {mam_SUITE,'-muc_prefs_set_cdata_request/1-fun-0-',2,
       [{file,"/home/circleci/project/big_tests/tests/mam_SUITE.erl"},
        {line,3407}]},
     {escalus_story,story,4,
       [{file,
          "/home/circleci/project/big_tests/_build/default/lib/escalus/src/escalus_story.erl"},
        {line,72}]},
     {muc_helper,story_with_room,4,
       [{file,"/home/circleci/project/big_tests/tests/muc_helper.erl"},
        {line,318}]},
     {test_server,ts_tc,3,[{file,"test_server.erl"},{line,1793}]},
     {test_server,run_test_case_eval1,6,
       [{file,"test_server.erl"},{line,1302}]},
     {test_server,run_test_case_eval,9,
       [{file,"test_server.erl"},{line,1234}]}]}}
```

[Report log](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4265/217970/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%40eb565858c58d.2024-04-30_06.48.16/big_tests.tests.mam_SUITE.logs/run.2024-04-30_06.48.18/mam_suite.muc_prefs_set_cdata_request.8324.html)
</details>
<details><summary>mam_SUITE:cassandra_muc_all:muc_prefs_cases:muc_prefs_set_request</summary>


```erlang
{error,
  {function_clause,
    [{mam_helper,parse_children_prefs_iq,
       [{xmlel,<<"error">>,
          [{<<"code">>,<<"501">>},{<<"type">>,<<"cancel">>}],
          [{xmlel,<<"feature-not-implemented">>,
             [{<<"xmlns">>,
             <<"urn:ietf:params:xml:ns:xmpp-stanzas">>}],
             []},
           {xmlel,<<"text">>,
             [{<<"xmlns">>,
             <<"urn:ietf:params:xml:ns:xmpp-stanzas">>}],
             [{xmlcdata,<<"From mod_mam_muc">>}]}]},
        {prefs_result_iq,undefined,[],[]}],
       []},
     {lists,foldl,3,[{file,"lists.erl"},{line,1594}]},
     {mam_SUITE,'-muc_prefs_set_request/1-fun-0-',3,
       [{file,"/home/circleci/project/big_tests/tests/mam_SUITE.erl"},
        {line,3335}]},
     {escalus_story,story,4,
       [{file,
          "/home/circleci/project/big_tests/_build/default/lib/escalus/src/escalus_story.erl"},
        {line,72}]},
     {muc_helper,story_with_room,4,
       [{file,"/home/circleci/project/big_tests/tests/muc_helper.erl"},
        {line,318}]},
     {test_server,ts_tc,3,[{file,"test_server.erl"},{line,1793}]},
     {test_server,run_test_case_eval1,6,
       [{file,"test_server.erl"},{line,1302}]},
     {test_server,run_test_case_eval,9,
       [{file,"test_server.erl"},{line,1234}]}]}}
```

[Report log](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4265/217970/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%40eb565858c58d.2024-04-30_06.48.16/big_tests.tests.mam_SUITE.logs/run.2024-04-30_06.48.18/mam_suite.muc_prefs_set_request.7938.html)
</details>
<details><summary>mam_SUITE:cassandra_muc_all:muc_prefs_cases:muc_messages_filtered_when_prefs_default_policy_is_roster</summary>


```erlang
{error,{test_case_failed,"ASSERT EQUAL\n\tExpected []\n\tValue [ok,ok,ok]\n"}}
```

[Report log](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4265/217970/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%40eb565858c58d.2024-04-30_06.48.16/big_tests.tests.mam_SUITE.logs/run.2024-04-30_06.48.18/mam_suite.muc_messages_filtered_when_prefs_default_policy_is_roster.8452.html)
</details>
<details><summary>mam_SUITE:cassandra_muc_all:muc_prefs_cases:muc_messages_filtered_when_prefs_default_policy_is_always</summary>


```erlang
{error,{test_case_failed,"ASSERT EQUAL\n\tExpected []\n\tValue [ok,ok,ok]\n"}}
```

[Report log](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4265/217970/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%40eb565858c58d.2024-04-30_06.48.16/big_tests.tests.mam_SUITE.logs/run.2024-04-30_06.48.18/mam_suite.muc_messages_filtered_when_prefs_default_policy_is_always.8194.html)
</details>
<details><summary>mam_SUITE:cassandra_muc_all:muc_prefs_cases:muc_messages_filtered_when_prefs_default_policy_is_never</summary>


```erlang
{error,{test_case_failed,"ASSERT EQUAL\n\tExpected []\n\tValue [ok,ok,ok,ok,ok,ok,ok]\n"}}
```

[Report log](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4265/217970/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%40eb565858c58d.2024-04-30_06.48.16/big_tests.tests.mam_SUITE.logs/run.2024-04-30_06.48.18/mam_suite.muc_messages_filtered_when_prefs_default_policy_is_never.8066.html)
</details>
<details><summary>mam_SUITE:elasticsearch_muc_all:muc_prefs_cases:muc_prefs_set_cdata_request</summary>


```erlang
{error,
  {function_clause,
    [{mam_helper,parse_children_prefs_iq,
       [{xmlel,<<"error">>,
          [{<<"code">>,<<"501">>},{<<"type">>,<<"cancel">>}],
          [{xmlel,<<"feature-not-implemented">>,
             [{<<"xmlns">>,
             <<"urn:ietf:params:xml:ns:xmpp-stanzas">>}],
             []},
           {xmlel,<<"text">>,
             [{<<"xmlns">>,
             <<"urn:ietf:params:xml:ns:xmpp-stanzas">>}],
             [{xmlcdata,<<"From mod_mam_muc">>}]}]},
        {prefs_result_iq,undefined,[],[]}],
       []},
     {lists,foldl,3,[{file,"lists.erl"},{line,1594}]},
     {mam_SUITE,'-muc_prefs_set_cdata_request/1-fun-0-',2,
       [{file,"/home/circleci/project/big_tests/tests/mam_SUITE.erl"},
        {line,3407}]},
     {escalus_story,story,4,
       [{file,
          "/home/circleci/project/big_tests/_build/default/lib/escalus/src/escalus_story.erl"},
        {line,72}]},
     {muc_helper,story_with_room,4,
       [{file,"/home/circleci/project/big_tests/tests/muc_helper.erl"},
        {line,318}]},
     {test_server,ts_tc,3,[{file,"test_server.erl"},{line,1793}]},
     {test_server,run_test_case_eval1,6,
       [{file,"test_server.erl"},{line,1302}]},
     {test_server,run_test_case_eval,9,
       [{file,"test_server.erl"},{line,1234}]}]}}
```

[Report log](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4265/217970/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%40eb565858c58d.2024-04-30_06.48.16/big_tests.tests.mam_SUITE.logs/run.2024-04-30_06.48.18/mam_suite.muc_prefs_set_cdata_request.16770.html)
</details>
<details><summary>mam_SUITE:elasticsearch_muc_all:muc_prefs_cases:muc_prefs_set_request</summary>


```erlang
{error,
  {function_clause,
    [{mam_helper,parse_children_prefs_iq,
       [{xmlel,<<"error">>,
          [{<<"code">>,<<"501">>},{<<"type">>,<<"cancel">>}],
          [{xmlel,<<"feature-not-implemented">>,
             [{<<"xmlns">>,
             <<"urn:ietf:params:xml:ns:xmpp-stanzas">>}],
             []},
           {xmlel,<<"text">>,
             [{<<"xmlns">>,
             <<"urn:ietf:params:xml:ns:xmpp-stanzas">>}],
             [{xmlcdata,<<"From mod_mam_muc">>}]}]},
        {prefs_result_iq,undefined,[],[]}],
       []},
     {lists,foldl,3,[{file,"lists.erl"},{line,1594}]},
     {mam_SUITE,'-muc_prefs_set_request/1-fun-0-',3,
       [{file,"/home/circleci/project/big_tests/tests/mam_SUITE.erl"},
        {line,3335}]},
     {escalus_story,story,4,
       [{file,
          "/home/circleci/project/big_tests/_build/default/lib/escalus/src/escalus_story.erl"},
        {line,72}]},
     {muc_helper,story_with_room,4,
       [{file,"/home/circleci/project/big_tests/tests/muc_helper.erl"},
        {line,318}]},
     {test_server,ts_tc,3,[{file,"test_server.erl"},{line,1793}]},
     {test_server,run_test_case_eval1,6,
       [{file,"test_server.erl"},{line,1302}]},
     {test_server,run_test_case_eval,9,
       [{file,"test_server.erl"},{line,1234}]}]}}
```

[Report log](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4265/217970/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%40eb565858c58d.2024-04-30_06.48.16/big_tests.tests.mam_SUITE.logs/run.2024-04-30_06.48.18/mam_suite.muc_prefs_set_request.16514.html)
</details>
<details><summary>mam_SUITE:elasticsearch_muc_all:muc_prefs_cases:muc_messages_filtered_when_prefs_default_policy_is_never</summary>


```erlang
{error,{test_case_failed,"ASSERT EQUAL\n\tExpected []\n\tValue [ok,ok,ok,ok,ok,ok,ok]\n"}}
```

[Report log](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4265/217970/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%40eb565858c58d.2024-04-30_06.48.16/big_tests.tests.mam_SUITE.logs/run.2024-04-30_06.48.18/mam_suite.muc_messages_filtered_when_prefs_default_policy_is_never.17026.html)
</details>
<details><summary>mam_SUITE:elasticsearch_muc_all:muc_prefs_cases:muc_messages_filtered_when_prefs_default_policy_is_roster</summary>


```erlang
{error,{test_case_failed,"ASSERT EQUAL\n\tExpected []\n\tValue [ok,ok,ok]\n"}}
```

[Report log](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4265/217970/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%40eb565858c58d.2024-04-30_06.48.16/big_tests.tests.mam_SUITE.logs/run.2024-04-30_06.48.18/mam_suite.muc_messages_filtered_when_prefs_default_policy_is_roster.16898.html)
</details>
<details><summary>mam_SUITE:elasticsearch_muc_all:muc_prefs_cases:muc_messages_filtered_when_prefs_default_policy_is_always</summary>


```erlang
{error,{test_case_failed,"ASSERT EQUAL\n\tExpected []\n\tValue [ok,ok,ok]\n"}}
```

[Report log](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4265/217970/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%40eb565858c58d.2024-04-30_06.48.16/big_tests.tests.mam_SUITE.logs/run.2024-04-30_06.48.18/mam_suite.muc_messages_filtered_when_prefs_default_policy_is_always.17282.html)
</details>

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/218184) / elasticsearch_and_cassandra_mnesia / e98370aa09d8daa158edf4f507d3bbdbcdc05dcb
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4265/218184/elasticsearch_and_cassandra_mnesia.26.1.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4265/218184/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%40ce7e2db75d86.2024-05-08_07.28.31/index.html)
**OK: 449** / Failed: 0 / **User-skipped: 41** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/218213) / elasticsearch_and_cassandra_mnesia / 7c0bb2e14cba8475fa43adc2cfddefcc68406f12
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4265/218213/elasticsearch_and_cassandra_mnesia.26.2.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4265/218213/elasticsearch_and_cassandra_mnesia.26.2.2-amd64/big/ct_run.test%40fde22f1a68b0.2024-05-08_08.42.04/index.html)
**OK: 457** / Failed: 0 / **User-skipped: 41** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/218414) / elasticsearch_and_cassandra_mnesia / 03ac017e759924f91d6c3fb142a5c235810927dc
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4265/218414/elasticsearch_and_cassandra_mnesia.26.2.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4265/218414/elasticsearch_and_cassandra_mnesia.26.2.2-amd64/big/ct_run.test%40189b2214b295.2024-05-09_08.05.29/index.html)
**OK: 457** / Failed: 0 / **User-skipped: 41** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/218433) / elasticsearch_and_cassandra_mnesia / 63008b4586ac2fa87e8c09153ab5009f10cb3eb7
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4265/218433/elasticsearch_and_cassandra_mnesia.26.2.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4265/218433/elasticsearch_and_cassandra_mnesia.26.2.2-amd64/big/ct_run.test%40f5cff0341d20.2024-05-09_08.07.15/index.html)
**OK: 457** / Failed: 0 / **User-skipped: 41** / Auto-skipped: 0

File: big_tests/tests/mam_SUITE.erl [hunk 1]
```diff
@@ -3517,6 +3569,36 @@ run_prefs_cases(DefaultPolicy, ConfigIn) ->
         end,
     escalus_fresh:story_with_config(ConfigIn, [{alice, 1}, {bob, 1}, {kate, 1}], F).
 
+muc_run_prefs_cases(DefaultPolicy, ConfigIn) ->
+    P = ?config(props, ConfigIn),
```
Comment (gustawlippa) [inline big_tests/tests/mam_SUITE.erl:3573]: I think this is not used

**Review by gustawlippa — CHANGES_REQUESTED** [review]

**Comment by gustawlippa** [comment]
Sorry, I clicked to early and didn't put a comment in review.

Great PR, nice to see the things that MIM had finally tested! I've requested changes for multiple small things that I've left comments for, but overall it's good 👍

File: big_tests/tests/mam_SUITE.erl [hunk 2]
```diff
@@ -3420,6 +3355,114 @@ prefs_set_cdata_request(Config) ->
         end,
     escalus_fresh:story(Config, [{alice, 1}], F).
 
+muc_prefs_set_request(ConfigIn) ->
+    F = fun(Config, Alice, _Bob) ->
+        %% Send
+        %%
+        %% <iq type='set' id='juliet2' to='room-alice'>
+        %%   <prefs xmlns='urn:xmpp:mam:1' default='roster'>
```
[the remark this answers is not in the document]
Comment (JanuszJakubiec) [reply big_tests/tests/mam_SUITE.erl:3363]: `*** User 2024-05-14 10:02:28.366 ***[🔗](https://github.com/esl/MongooseIM/pull/4265/files#e-175)
alice_muc_prefs_set_request_13@localhost/res1 out
<iq type='set' id='b18e5acf33e5dca67f962533a5ac9210' to='room-alice_muc_prefs_set_request_13@muc.localhost'>
	<prefs xmlns='urn:xmpp:mam:1' default='roster'>
		<always>
			<jid>romeo@montague.net</jid>
		</always>
		<never>
			<jid>montague@montague.net</jid>
		</never>
	</prefs>
</iq>`

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/218600) / elasticsearch_and_cassandra_mnesia / 02a82db369c10e9946f357de56e137a174e705b1
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4265/218600/elasticsearch_and_cassandra_mnesia.26.2.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4265/218600/elasticsearch_and_cassandra_mnesia.26.2.2-amd64/big/ct_run.test%40ed83b7d2500d.2024-05-14_13.16.54/index.html)
**OK: 457** / Failed: 0 / **User-skipped: 41** / Auto-skipped: 0

File: big_tests/tests/mam_SUITE.erl (hunk 2 above)
[the remark this answers is not in the document]
Comment (gustawlippa) [reply big_tests/tests/mam_SUITE.erl:3363]: Huh, this is from the test, right? I was looking at the XEP: https://xmpp.org/extensions/xep-0441.html#example-4

File: big_tests/tests/mam_SUITE.erl [hunk 3]
```diff
@@ -905,8 +878,7 @@ maybe_destroy_room(CaseName, Config) ->
 
 all_cases_with_room() ->
     muc_cases_with_room() ++ muc_fetch_specific_msgs_cases() ++ muc_configurable_archiveid_cases() ++
-        muc_stanzaid_cases() ++ muc_retract_cases() ++ muc_metadata_cases() ++
-        muc_text_search_cases() ++ [muc_message_dropped].
+        muc_stanzaid_cases() ++ muc_retract_cases() ++ muc_metadata_cases() ++ muc_text_search_cases().
```
[the remark this answers is not in the document]
Comment (gustawlippa) [reply big_tests/tests/mam_SUITE.erl:881]: Hmm, that's surprising that there is some other Alice 😄
But I understand that this way it's easier while setting the prefs. Maybe it's something we should revisit in Escalus to make more sense.

**Review by gustawlippa — APPROVED** [review]

**Pull request merged**